agent/provider/run-ya-provider.sh

```diff
@@ -0,0 +1,221 @@
+#!/bin/bash
+
+TESTNET=testnet
+
+usage() {
+  cat 1>&2 <<EOF
+Runs Yagna Service and ya-provider from given deb.
+Installs VM and WASI runtimes from github.
+
+Usage:
+    $(basename $0) [-t <testnet>] <yagna-deb>
+
+Options:
+  -t <string>  sets testnet name ('$TESTNET' by default)
+
+Steps:
+    1. Downloads and installs VM and WASI runtimes.
+    2. Installs and (re)starts Yagna Service (logs output to ./yagna.log).
+    3. Configures and runs ya-provider (logs output to ./ya-provider.log).
+EOF
+}
+
+#############################################
+## helpers, adapted from from sh.rustup.rs ##
+
+get_install_dir() {
+  basename $1 | sed -e 's/\(_amd64\)\?.deb$//' -e 's/^\(yagna\)\?/testnet/'
+}
+
+say() {
+    printf 'yagna-up [%s]: %s\n' "$(date +'%Y-%m-%d %H:%M:%S')" "$@"
+}
+
+err() {
+    say "$1" >&2
+    exit 1
+}
+
+need_cmd() {
+    if ! check_cmd "$1"; then
+        err "need '$1' (command not found)"
+    fi
+}
+
+check_cmd() {
+    command -v "$1" > /dev/null 2>&1
+}
+
+# Run a command that should never fail.
+ensure() {
+    if ! "$@"; then err "command failed: $*"; fi
+}
+
+# To indicate that commands' results are being intentionally ignored.
+ignore() {
+    "$@"
+}
+
+presets_json() {
+    cat << EOF
+{
+  "active": [
+    "vm",
+    "wasmtime"
+  ],
+  "presets": [
+    {
+      "name": "vm",
+      "exeunit-name": "vm",
+      "pricing-model": "linear",
+      "usage-coeffs": {
+        "duration": 0.001,
+        "cpu": 0.002,
+        "initial": 0.0
+      }
+    },
+    {
+      "name": "wasmtime",
+      "exeunit-name": "wasmtime",
+      "pricing-model": "linear",
+      "usage-coeffs": {
+        "initial": 1.0,
+        "cpu": 1.0,
+        "duration": 0.1
+      }
+    }
+  ]
+}
+EOF
+}
+
+#################
+###  m a i n  ###
+#################
+run() {
+    need_cmd mkdir
+    need_cmd sed
+    need_cmd wget
+    need_cmd dpkg-deb
+
+    local testnet="testnet"
+    local deb_file
+
+    while [[ "$#" -gt 0 ]]; do
+        case $1 in
+            -h|--help) usage; exit 0 ;;
+            -t) testnet="$2"; shift; shift ;;
+            -*) usage; echo; err "unexpected args '$*'"; ;;
+            *) deb_file="$1"; shift; break ;;
+        esac
+    done
+
+    if [[ -z "$deb_file" ]]; then
+      usage
+      echo
+      err "Please specify deb file to be used."
+    fi
+
+    if [[ "$#" -gt 0 ]]; then
+      usage
+      echo
+      err "unexpected args '$*'"
+    fi
+
+    local install_dir="$HOME/$(get_install_dir $deb_file)"
+
+    if [[ ! -d "$install_dir" ]]; then
+        say "Creating install dir $install_dir"
+        ensure mkdir "$install_dir"
+    else
+        say "Using install dir $install_dir"
+    fi
+
+    cd "$install_dir"
+
+    if [[ ! -f usr/lib/yagna/plugins/ya-runtime-wasi.json ]]; then
```

Comment:
Perhaps leave a comment here about this path being intentional, as it seems to be confusing (I had the same question about this part).